executors/rust/1.5/src/main.rs

```diff
@@ -0,0 +1,144 @@
+    let _supported_test_map = HashMap::from([
+        ("collation_short".to_string(), run_collation_test), // TODO: ,("number_fmt".to_string(), run_numberformat_test)
+    ]);
+
+    // TODO: supported_test_map to call the functions.
+
+    // TODO: Handle problem with
+    // Error: Custom { kind: INvalidData, error: Error{"unexpected end of hex escape"
+    // As in collation 000144, 0998, 0142
+    let mut buffer = String::new();
+
+    loop {
+        let buffer_size = io::stdin().read_line(&mut buffer)?;
+        if buffer_size == 0 {
+            break;
+        }
+        if buffer.starts_with("#EXIT") {
+            break;
+        }
+        if buffer.starts_with("#TESTS") {
+            // Returns JSON list of supported tests.
+            // TODO: let mut test_vec : Vec<&str> = supported_test_map.into_keys().collect();
+            let json_result = json!(
+                { "supported_tests": [
+                    "collation_short",
+                    "number_fmt",
+                    "decimal_fmt",
+                    "likelysubtags",
+                    "list_fmt",
+                    "plural_rules"
+                ] }
+            );
+            println!("{}", json_result);
+        }
+
+        if buffer.starts_with("#VERSION") {
+            let json_result = json!(
+            {
+                "platform": "ICU4X",
+                "platformVersion": std::env!("CONFORMANCE_ICU4X_VERSION"),
+                "icuVersion": std::env!("CONFORMANCE_ICU_VERSION"),
+                "cldrVersion": std::env!("CONFORMANCE_CLDR_VERSION"),
+            });
+            println!("{}", json_result);
+        } else {
+            // Expecting test information as JSON data in a single line.
+
+            //  https://stackoverflow.com/questions/30292752/how-do-i-parse-a-json-file
+            let json_info: Value = serde_json::from_str(&buffer)?;
+
+            let test_type: &str = json_info["test_type"].as_str().unwrap();
+            let label: &str = json_info["label"].as_str().unwrap();
+
+            // TODO!!! : supported_test_map to call the functions.
+            let json_result = if test_type == "collation_short" {
+                run_collation_test(&json_info)
+            } else if (test_type == "decimal_fmt") || (test_type == "number_fmt") {
+                run_numberformat_test(&json_info)
+            } else if (test_type == "display_names")
+                || (test_type == "language_display_name")
+                || (test_type == "lang_names")
+            {
+                run_language_name_test(&json_info)
+            } else if test_type == "likely_subtags" {
+                run_likelysubtags_test(&json_info)
+            } else if test_type == "list_fmt" {
+                run_list_fmt_test(&json_info)
+            } else if test_type == "plural_rules" {
+                run_plural_rules_test(&json_info)
+            } else {
+                Err(test_type.to_string())
+            };
```

Comment:
This kind of code is highly likely to go stale. It would be _nice_ to keep it in sync across versions. I don't immediately see why it needs to be different between 1.4 and 1.5?